### PR TITLE
Add fields to MessageAttachment

### PR DESCRIPTION
**Pull request by patrickxb**
r? @maxtaco @mmaxim

### CODE REVIEW

File: protocol/avdl/chat1/local.avdl [hunk 1]
```diff
@@ -22,7 +22,10 @@ protocol local {
   }
 
   record MessageAttachment {
-    string path;
+    string storageType;  // where the attachment object is stored (i.e. s3)
```
Comment (maxtaco) [inline protocol/avdl/chat1/local.avdl]: do you think we'll want to version the attachment structure?  What if version 2 is just to KBFS and therefore doesn't need a key?

Comment (patrickxb) [inline protocol/avdl/chat1/local.avdl]: Good point.  Could then remove the encryption version int as that will be implied by the version of the attachment struct.

Comment (maxtaco) [inline protocol/avdl/chat1/local.avdl]: I wonder if `MessageAttachment` should be a `variant`.  We can alternatively have `ATTACHMENT` v `ATTACHMENT_V2` in the variant one level up

**Comment by mmaxim** [comment]
LGTM!

**Pull request merged**